h2o-algos/src/test/java/hex/splitframe/ShuffleSplitFrameTest.java

```diff
@@ -115,5 +113,82 @@ static void testScenario(Frame f, String[] expValues, MRTask chunkAssertions) {
     return ary;
   }
 
+
+  @Test
+  public void splitRatioIsWorkingCorrectly() {
+    Frame leaderboardFrame = null;
+
+    try {
+      leaderboardFrame = new TestFrameBuilder()
+              .withName("leaderboardFrame")
+              .withColNames("ColA", "ColB", "ColC")
+              .withVecTypes(Vec.T_CAT, Vec.T_CAT, Vec.T_CAT)
+              .withDataForCol(0, ar("a", "b", "b", "b", "a", "a"))
+              .withDataForCol(1, ar("yellow", "blue", "green", "red", "purple", "orange"))
+              .withDataForCol(2, ar("2", "6", "6", "6", "6", "2"))
+              .build();
+
+      double[] splitRatio = {0.8, 0.2};
+      Key[] keysForSplits = {Key.make("key_" + leaderboardFrame._key), Key.make("key2_" + leaderboardFrame._key)};
+      Frame[] splits = ShuffleSplitFrame.shuffleSplitFrame(leaderboardFrame, keysForSplits, splitRatio, 1234);
+      System.out.println("Number of row in original frame: " + leaderboardFrame.numRows());
+      assertEquals(5, splits[0].numRows());
+      assertEquals(1, splits[1].numRows());
+
+    } finally {
+      leaderboardFrame.delete();
+    }
+  }
+
+  @Test
+  public void splitRatioIsWorkingCorrectlyOnBigData() {
+    Frame leaderboardFrame = null;
+
+    try {
+      leaderboardFrame = new TestFrameBuilder()
+              .withName("leaderboardFrame")
+              .withColNames("ColA")
+              .withVecTypes(Vec.T_NUM)
+              .withRandomDoubleDataForCol(0, 1000, 20, 90, 1234L)
+              .build();
+
+      double[] splitRatio = {0.8, 0.2};
+      Key[] keysForSplits = {Key.make("key_" + leaderboardFrame._key), Key.make("key2_" + leaderboardFrame._key)};
+      Frame[] splits = ShuffleSplitFrame.shuffleSplitFrame(leaderboardFrame, keysForSplits, splitRatio, 1234);
+      System.out.println("Number of row in original frame: " + leaderboardFrame.numRows());
+      assertEquals(800, splits[0].numRows(), 10);
+      assertEquals(200, splits[1].numRows(), 10);
+
+    } finally {
+      leaderboardFrame.delete();
+    }
+  }
+
+  @Test
+  public void numberOfPartitionsIsMoreImportantThenCorrectRation() {
+    Frame leaderboardFrame = null;
+
+    try {
+      leaderboardFrame = new TestFrameBuilder()
+              .withName("leaderboardFrame")
+              .withColNames("ColA", "ColB", "ColC")
+              .withVecTypes(Vec.T_CAT, Vec.T_CAT, Vec.T_CAT)
+              .withDataForCol(0, ar("a", "b"))
+              .withDataForCol(1, ar("yellow", "blue"))
+              .withDataForCol(2, ar("2", "6"))
+              .build();
+
+      double[] splitRatio = {0.9, 0.1};
+      Key[] keysForSplits = {Key.make("key_" + leaderboardFrame._key), Key.make("key2_" + leaderboardFrame._key)};
+      Frame[] splits = ShuffleSplitFrame.shuffleSplitFrame(leaderboardFrame, keysForSplits, splitRatio, 1234);
+      System.out.println("Number of row in original frame: " + leaderboardFrame.numRows());
+      assertEquals(1, splits[0].numRows());
```

Comment:
What do you propose the result should be?

Reply:
Lets say for the edge case of 2 rows and ratios [0.9, 0.1]  I would expect two splits with one row in each because it is I believe a higher priority to fulfil implicit user's request of getting 2 non-empty splits . But the main point is that we should not end up with empty split when let's say I'm splitting 6-rows dataset with [0.8, 0.2] ratios. In current implementation it is quite often the case.